src/file/reader.rs

```diff
@@ -197,7 +233,7 @@ impl SerializedFileReader {
   }
 }
 
-impl FileReader for SerializedFileReader {
+impl<R: 'static + ParquetReader> FileReader for SerializedFileReader<R> {
```

Comment:
Does usage of `'static` fix the example in the PR description?

Reply:
No, is what prevents that it works. Not sure if it's due to boxing or DataType. `'static` implies that outlives any other lifetime, which does not happen on the example (as the vector will be dropped when going out of the scope).